UnityProject/Assets/Scripts/Systems/Antagonists/Antags/Blob/BlobPlayer.cs

```diff
@@ -0,0 +1,1266 @@
+
+			if (!excludeNodes)
+			{
+				foreach (var node in nodeBlobs)
+				{
+					if(node == null) continue;
+
+					specialisedBlobs.Add(node.location);
+				}
+			}
+
+			if (!excludeFactory)
+			{
+				foreach (var factory in factoryBlobs)
+				{
+					if(factory.Key == null) continue;
+
+					specialisedBlobs.Add(factory.Key.GetComponent<BlobStructure>().location);
+				}
+			}
+
+			if (!excludeResource)
+			{
+				foreach (var resourceBlob in resourceBlobs)
+				{
+					if(resourceBlob == null) continue;
+
+					specialisedBlobs.Add(resourceBlob.GetComponent<BlobStructure>().location);
+				}
+			}
+
+			foreach (var blob in specialisedBlobs)
+			{
+				if (Vector3Int.Distance(blob, worldPos) <= buildDistanceLimit)
+				{
+					//A blob is too close
+					return false;
+				}
+			}
+
+			return true;
+		}
+
+		#endregion
+
+		#region Effects
+
+		private void PlayAttackEffect(Vector3 worldPos)
+		{
+			var result = Spawn.ServerPrefab(attackEffect, worldPos, gameObject.transform.parent);
+
+			if (!result.Successful) return;
+
+			StartCoroutine(DespawnEffect(result.GameObject));
+		}
+
+		private IEnumerator DespawnEffect(GameObject effect)
+		{
+			yield return WaitFor.Seconds(1f);
+
+			Despawn.ServerSingle(effect);
+		}
+
+		#endregion
+
+		#region PlaceStrong/Reflective
+
+		[Command]
+		public void CmdTryPlaceStrongReflective(Vector3Int worldPos)
+		{
+			worldPos.z = 0;
+
+			if (!ValidateAction(worldPos)) return;
+
+			if (blobTiles.TryGetValue(worldPos, out var blob) && blob != null)
+			{
+				//Try place strong
+				if (blob != null && blob.isNormal)
+				{
+					PlaceStrongReflective(blobStrongPrefab, blob, strongBlobCost, worldPos, "You grow a strong blob, you can now mutate to a reflective.");
+					return;
+				}
+
+				//Try place reflective
+				if (blob != null && blob.isStrong)
+				{
+					PlaceStrongReflective(blobReflectivePrefab, blob, reflectiveBlobCost, worldPos, "You grow a reflective blob.");
+					return;
+				}
+			}
+
+			//No normal blob at tile
+			Chat.AddExamineMsgFromServer(gameObject, "You need to place a normal blob first");
+		}
+
+		private void PlaceStrongReflective(GameObject prefab, BlobStructure originalBlob, int cost, Vector3Int worldPos, string msg)
+		{
+			if (!ValidateCost(cost, prefab)) return;
+
+			var result = Spawn.ServerPrefab(prefab, worldPos, gameObject.transform);
+
+			if (!result.Successful) return;
+
+			Chat.AddExamineMsgFromServer(gameObject, msg);
+
+			Despawn.ServerSingle(originalBlob.gameObject);
+			var structure = result.GameObject.GetComponent<BlobStructure>();
+			SetLightAndColor(structure);
+			blobTiles[worldPos] = structure;
+			AddNonSpaceBlob(result.GameObject);
+		}
+
+		#endregion
+
+		#region PlaceOther
+
+		[Command]
+		public void CmdTryPlaceOther(Vector3Int worldPos, BlobConstructs blobConstructs)
+		{
+			worldPos.z = 0;
+
+			if (!ValidateAction(worldPos)) return;
+
+			GameObject prefab = null;
+
+			var cost = 0;
+
+			switch (blobConstructs)
```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6357029)